slip_estimate_node:
  ros__parameters:

    # Kalman filter model
    Qx: 0.05 # Process uncertainty for x
    Qxdot: 5.0 # Process uncertainty for xdot
    Qax: 0.01 # Process uncertainty for xddot
    Qy: 0.05 # Process uncertainty for y
    Qydot: 5.0 # Process uncertainty for ydot
    Qay: 0.01 # Process uncertainty for yddot
    # Rx: 1.0 # Measurement uncertainty for x
    Rxdot: 1.0 # Measurement uncertainty for xdot
    Rax: 1.0 # Measurement uncertainty for xddot
    # Ry: 1.0 # Measurement uncertainty for y
    Rydot: 1.0 # Measurement uncertainty for ydot
    Ray: 1.0 # Measurement uncertainty for yddot
    Px: 0.1 # Initial estimated uncertainty in x
    Pxdot: 0.1 # Initial estimated uncertainty for xdot
    Pax: 0.1 # Initial estimated uncertainty for xddot
    Py: 0.1 # Initial estimated uncertainty for y
    Pydot: 0.1 # Initial estimated uncertainty for ydot
    Pay: 0.1 # Initial estimated uncertainty for yddot
    kf_dt: 0.2 # Filter time step, assumed to be constant

    vel_kf_window_size: 7 # Number of points to use in moving average of velocity estimate

    # Slip estimation thresholds
    nonzero_slip_thresh_wheel_ms: 0.1 # Only calculate slip when wheel speed is ABOVE this threshold [m/s]
    nonzero_slip_thresh_vehicle_ms: 0.07 # Only calculate slip if vehicle velocity is BELOW this threshold [m/s]

    slip_latch_thresh_ms: 0.75 # Threshold to latch slip above
    slip_velocity_latch_release_ms: 0.06 # Release slip latch once estimated vehicle body speed is above this threshold [m/s]
    
    slip_window_size: 12 # Number of points to use in moving average of slip estimate
    